test_genie_python_dae.py

```diff
@@ -310,6 +310,53 @@ def test_GIVEN_change_tables_called_WHEN_filenames_are_not_raw_strings_and_with_
             spectra=spectra
         )
 
+    def test_GIVEN_change_number_soft_periods_called_WHEN_new_value_normal_THEN_change_successful(self):
+        set_genie_python_raises_exceptions(True)
+
+        g.change_number_soft_periods(30)
+        sleep(10)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        set_genie_python_raises_exceptions(False)
+
+    def test_GIVEN_change_number_soft_periods_called_WHEN_new_value_too_big_for_DAE_hardware_THEN_raise_exception_to_console(self):
+        set_genie_python_raises_exceptions(True)
+
+        g.change_number_soft_periods(30)
+        sleep(10)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        self.assertRaises(IOError, g.change_number_soft_periods, 1000000)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        set_genie_python_raises_exceptions(False)
+
+    @parameterized.expand(parameterized_list([1, 2, 6, 9, 10]))
```

Comment:
we seem to be not using the 0th indexed tuple value from each tuple. Why not just pass a list of lists?

Reply:
It is just easier this way and looks less messy